jvector-base/src/main/java/io/github/jbellis/jvector/pq/KMeansPlusPlusClusterer.java

```diff
@@ -143,18 +143,24 @@ private float[][] chooseInitialCentroids(float[][] points) {
 
     /**
      * Assigns points to the nearest cluster.  The results are stored as ordinals in `assignments`
+     *
+     * @param firstRun whether this is the first assignment of points, which means the previous assignment is meaningless
+     * @return the number of points that changed clusters
      */
-    private int assignPointsToClusters() {
+    private int assignPointsToClusters(boolean firstRun) {
```

Comment:
i don't love "flag parameters" that change behavior, could we split this into assignInitialPoints and assignIteratedPoints (names subject to change)?